src/mimefactory.rs

```diff
@@ -2363,7 +2346,8 @@ mod tests {
             .await
             .unwrap();
 
-        // send message to bob: that should get multipart/mixed because of the avatar moved to inner header;
+        // send message to bob: that should get multipart/signed.
+        // `Subject:` is protected by copying it.
```

Comment:
Not sure copying Subject into the IMF section is really needed. Are there any MUAs not parsing it from the "multipart/signed" part? This is minor though